src/serial/remote/protocol_v0.rs

```diff
@@ -169,6 +171,59 @@ impl BmdRemoteProtocol for RemoteV0
 	{
 		Err(eyre!("Not supported"))
 	}
+
+	fn get_target_voltage(&self) -> Result<f32>
+	{
+		self.interface().buffer_write(RemoteCommands::TARGET_VOLTAGE)?;
+		let buffer = self.interface().buffer_read()?;
+
+		if buffer.is_empty() || buffer.as_bytes()[0] != REMOTE_RESP_OK {
+			return Err(eyre!("Target voltage request failed"));
+		}
+
+		if buffer.len() < 2 {
+			return Err(eyre!("Target voltage response is too short"));
+		}
+
+		let value = buffer
+			.get(1..buffer.len().saturating_sub(1))
+			.expect("Should have some value");
+		value
+			.parse::<f32>()
+			.map_err(|e| eyre!("Can't parse target voltage value to a float, input {}, reason: {}", value, e))
+	}
+
+	fn get_srst_val(&self) -> Result<bool>
+	{
+		self.interface().buffer_write(RemoteCommands::NRST_GET)?;
+		let buffer = self.interface().buffer_read()?;
+
+		if buffer.is_empty() || buffer.as_bytes()[0] != REMOTE_RESP_OK {
+			return Err(eyre!("srst value request failed"));
+		}
+
+		if buffer.len() < 2 {
+			return Err(eyre!("srst value response is too short"));
+		}
+
+		Ok(buffer.as_bytes()[1] == b'1')
+	}
+
+	fn get_target_supply_power(&self) -> Result<bool>
+	{
+		self.interface().buffer_write(RemoteCommands::PWR_GET)?;
+		let buffer = self.interface().buffer_read()?;
+
+		if buffer.is_empty() || buffer.as_bytes()[0] != REMOTE_RESP_OK {
+			return Err(eyre!("remote_target_get_power value request failed"));
+		}
+
+		if buffer.len() < 2 {
+			return Err(eyre!("remote_target_get_power value response is too short"));
+		}
+
+		Ok(buffer.as_bytes()[1] == b'1')
+	}
```

Comment:
This duplicates `get_target_power_state` which is also the better name for what this is and does.